app/routes.py

```diff
@@ -1,2 +1,238 @@
+        return jsonify(None), 404
+    form_data = request.get_json()
+
+    task.title = form_data["title"]
+    task.description = form_data["description"]
+    task.iscomplete = None
+
+    db.session.commit()
+
+    task_response = {"task": task.create_dict()}
+    return jsonify(task_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        return jsonify(None), 404
+
+    db.session.delete(task)
+    db.session.commit()
+
+    delete_message = {
+        "details": f'Task {task.task_id} "{task.title}" successfully deleted'}
+
+    return jsonify(delete_message), 200
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def task_complete(task_id):
+    if request.method == "PATCH":
+        task = Task.query.get(task_id)
+        if not task:
+            return jsonify(None), 404
+
+        task.completed_at = datetime.datetime.now()
+        db.session.commit()
+
+        task.send_task_complete_slack_message()
+
+        task_response = {"task": task.create_dict()}
+        return jsonify(task_response), 200
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def task_incomplete(task_id):
+    if request.method == "PATCH":
+        task = Task.query.get(task_id)
+        if not task:
+            return jsonify(None), 404
+
+        task.completed_at = None
+        db.session.commit()
+
+        task_response = {"task": task.create_dict()}
+        return jsonify(task_response), 200
+
+
+#
+# Start Goals Routes
+#
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    title_from_url = request.args.get("title")
+
+    goals = Goal.goal_arguments(title_from_url)
+    goals_response = []
+    goals_response = [goal.create_dict() for goal in goals]
+
+    if not goals_response:
+        goals = Goal.query.all()
+        goals_response = [goal.create_dict() for goal in goals]
+
+    return jsonify(goals_response)
+
+
+@goals_bp.route("", methods=["POST"])
+def new_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        error_dict = {"details": "Invalid data"}
+        return jsonify(error_dict), 400
+
+    new_goal = Goal(title=request_body["title"])
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    goal_response = {"goal": new_goal.create_dict()}
+    return jsonify(goal_response), 201
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_individual_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return jsonify(None), 404
+
+    if request.method == "GET":
+        goal_response = {"goal": goal.create_dict()}
+        return jsonify(goal_response), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["PUT"])
+def replace_individual_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return jsonify(None), 404
+    form_data = request.get_json()
+
+    goal.title = form_data["title"]
```

Comment:
Make sure to at least confirm that the key exists in the request data before accessing the dictionary, or we might end up with a KeyError